File: doc/src/codes/Users_Manual/gain/transform_0.cs
using AUTD3Sharp;
using AUTD3Sharp.Gain;

var g = new Uniform(EmitIntensity.Max).WithTransform((dev, tr, d) =>
{
    d.Intensity = new EmitIntensity((byte)(d.Intensity.Value / 2));
    d.Phase = Phase.FromRad(d.Phase.Radian + AUTD3.Pi);
    return d;
});

File: doc/src/codes/Users_Manual/gain/transform_0.rs
# extern crate autd3;
use autd3::prelude::*;

# #[allow(unused_variables)]
# fn main() {
let g = Uniform::new(EmitIntensity::MAX).with_transform(|dev, tr, d| Drive {
    intensity: EmitIntensity::new(d.intensity.value() / 2),
    phase: Phase::from_rad(d.phase.radian() + PI),
});
# }